go/libraries/doltcore/sqle/dfunctions/dolt_push.go

```diff
@@ -91,7 +91,9 @@ func DoDoltPush(ctx *sql.Context, args []string) (int, error) {
 		return cmdFailure, err
 	}
 
-	opts, err := env.NewPushOpts(ctx, apr, dbData.Rsr, dbData.Ddb, apr.Contains(cli.ForceFlag), apr.Contains(cli.SetUpstreamFlag))
+	autoSetUpRemote := loadConfig(ctx).GetStringOrDefault(env.PushAutoSetupRemote, "false")
+
+	opts, err := env.NewPushOpts(ctx, apr, dbData.Rsr, dbData.Ddb, apr.Contains(cli.ForceFlag), apr.Contains(cli.SetUpstreamFlag), autoSetUpRemote == "true")
```

Comment:
(minor) does `GetStringOrDefault` convert the config value to lower case already? You might consider adding a test for `True` (in addition to `true`). Not a big deal, but seems like a reasonable place to expect case not to matter.